test/test_functions.py

```diff
@@ -0,0 +1,352 @@
+
+from collections import namedtuple
+from decimal import Decimal
+import pytest
+import random
+
+from pyrgg.functions import (
+    get_precision,
+    is_float,
+    weight_str_to_number,
+    convert_bytes,
+    filesize,
+    time_convert,
+    input_filter,
+    sign_gen,
+    branch_gen,
+    edge_gen,
+)
+
+
+@pytest.mark.parametrize("number, expected", [
+    (1.2, 1),
+    (1.12, 2),
+    (1.0002, 4),
+    (10, 0),
+    (10.1, 1),
+    (10.02, 2),
+    (10.20, 1),  # or should it be 2?
```

Comment:
I think it should be 2 because `10.20` is more precise than `10.2`(second could be any number between `10.2000...` to `10.2999...`.
what do you think @sepandhaghighi ?